terraform/modules/github-ci-bootstrap/README.md

```diff
@@ -181,16 +231,36 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       
-      - name: Authenticate to Google Cloud
+      - name: Authenticate to Google Cloud (Write-Enabled Branches)
+        if: contains(fromJSON('["refs/heads/main", "refs/heads/master"]'), github.ref)
         uses: google-github-actions/auth@v2
         with:
           workload_identity_provider: ${{ outputs.workload_identity_provider }}
           service_account: ${{ outputs.service_account_email }}
           
+      - name: Authenticate to Google Cloud (Read-Only - Any Branch)
```

Comment:
Does this block not need an `if` as well?  Otherwise, won't both blocks execute on `main`, and step on each other?